### ISSUE TITLE
Failed to remove instance as device is busy

### ISSUE THREAD

**Issue by krystism** [docker/docker#10174]
# Description of problem
I want to remove all my instances as follows:
```
docker rm -f  `docker ps -a -q`
```
However, it seems that I  **FAILED TO REMOVE ANY ONE**, the output is:
```
Error response from daemon: Cannot destroy container b719f5700c1c: Driver devicemapper failed to remove root filesystem b719f5700c1c3f17e6f61506de1326e736d63aa692150957baa82cb882ab02ed: Device is Busy
...
...
FATA[0137] Error: failed to remove one or more containers
```
But when I exec
``` 
docker ps -a
```
I can't find any instance from the output, it seems to remove them successfully! 
But That's just a *pseudomorph*，the docker  lies to me! when I restart the docker daemon, All these guys emerge like the ghost.
# Environment details
The docker is running on an instance of openstack using kvm.
```
krystism@docker:~$ docker version
Client version: 1.4.1
Client API version: 1.16
Go version (client): go1.3.3
Git commit (client): 5bc2ff8
OS/Arch (client): linux/amd64
Server version: 1.4.1
Server API version: 1.16
Go version (server): go1.3.3
Git commit (server): 5bc2ff8

krystism@docker:~$ docker info
Containers: 13
Images: 351
Storage Driver: devicemapper
 Pool Name: docker-253:1-131744-pool
 Pool Blocksize: 65.54 kB
 Data file: /var/lib/docker/devicemapper/devicemapper/data
 Metadata file: /var/lib/docker/devicemapper/devicemapper/metadata
 Data Space Used: 10.35 GB
 Data Space Total: 107.4 GB
 Metadata Space Used: 18.11 MB
 Metadata Space Total: 2.147 GB
 Library Version: 1.02.82-git (2013-10-04)
Execution Driver: native-0.2
Kernel Version: 3.13.0-35-generic
Operating System: Ubuntu 14.04.1 LTS
CPUs: 2
Total Memory: 3.861 GiB
Name: docker
ID: ALD7:CBPH:KKHE:KDBQ:GWAF:LXRR:E6Z5:6M5S:3RGY:MA4D:NWKB:QIK7
WARNING: No swap limit support

krystism@docker:~$ uname -a
Linux docker 3.13.0-35-generic #62-Ubuntu SMP Fri Aug 15 01:58:42 UTC 2014 x86_64 x86_64 x86_64 GNU/Linux
```
# Steps to Reproduce
1. Create some containers.
2. Try to remove them.
3. Restart the docker daemon.
4. Exec "docker ps -a".

Actual Results: All the removed container emerge again.
Expected Results: Should not show any instance.

**Comment by thaJeztah** [comment]
dup of https://github.com/docker/docker/issues/5771 or https://github.com/docker/docker/issues/5684?

**Comment by cpuguy83** [comment]
This might be fixed on master by #9886

**Comment by unclejack** [comment]
This is a duplicate of #5684 and #4036.

This particular error will be fixed by #10225. Other errors seen in #4036 have been fixed by #10195.

I'm going to close this now.
